src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -35,6 +36,21 @@ void regclass_Core(py::module m) {
         py::arg("config"),
         py::arg("device_name") = "");
 
+    cls.def(
+        "set_property",
+        [](ov::Core& self, const std::map<std::string, std::string>& properties) {
+            self.set_property({properties.begin(), properties.end()});
+        },
+        py::arg("properties"));
+
+    cls.def(
+        "set_property",
+        [](ov::Core& self, const std::string& device_name, const std::map<std::string, std::string>& properties) {
+            self.set_property(device_name, {properties.begin(), properties.end()});
```

Comment:
Let's think about reusing `PyAny` wrapper class to replace `std::map<std::string, std::string>` with AnyMap-like construct of `std::map<std::string, ov::Any>`.

Reply:
done